hermes/bin/src/pool.rs

```diff
@@ -0,0 +1,98 @@
+//! Global thread pool for parallel WASM module execution.
+//!
+//! This module initializes a global Rayon thread pool
+//! for running WASM modules in parallel. The pool uses
+//! work-stealing for efficient load balancing across
+//! available threads. To avoid saturating the system,
+//! the pool leaves a small margin of CPU capacity
+//! unused — typically 2 threads are reserved for the
+//! main application flow and event loop.
+//!
+//! Each event may execute multiple WASM modules
+//! concurrently within this pool.
+
+use std::{
+    sync::{atomic::AtomicUsize, Condvar, Mutex, OnceLock},
+    thread::available_parallelism,
+};
+
+use anyhow::{Context, Result};
+use rayon::ThreadPoolBuilder;
+
+/// Global counter of currently running tasks.
+static TASK_COUNTER: AtomicUsize = AtomicUsize::new(0);
+
+/// Synchronization primitives for waiting until all tasks finish.
+static TASK_WAIT: OnceLock<(Mutex<()>, Condvar)> = OnceLock::new();
+
+/// Get a reference to the global `(Mutex, Condvar)` tuple,
+/// initializing it if necessary.
+fn get_task_wait() -> &'static (Mutex<()>, Condvar) {
+    TASK_WAIT.get_or_init(|| (Mutex::new(()), Condvar::new()))
+}
```

Comment:
Isn't `OnceLock`excessive?

```suggestion
/// Synchronization primitives for waiting until all tasks finish.
static TASK_WAIT: (Mutex<()>, Condvar) = (Mutex::new(), Condvar::new());

/// Get a reference to the global `(Mutex, Condvar)` tuple,
/// initializing it if necessary.
fn get_task_wait() -> &'static (Mutex<()>, Condvar) {
    &TASK_WAIT
}
```

Reply:
Agreed, this could be simplified